lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -160,6 +160,9 @@ export abstract class ClientApplication {
                 // Navigate to page that initiated the redirect request
                 BrowserUtils.navigateWindow(loginRequestUrl, true);
             }
+
+            // Due to navigation this should not resolve, delay to ensure code dependant on handleRedirectPromise doesn't run before navigation
+            return new Promise(resolve => setTimeout(resolve, 5000, null));
```

Comment:
Not sure how I feel about an arbitrary delay like this. This would impact all customers right? Even those not calling `loginRedirect` after `handleRedirectPromise`?

Reply:
Ah, this only runs in scenarios where we are redirecting again, correct?

Reply:
Can the `5000` be put into a constant?

Reply:
This line is only hit when we need to go from redirectUri -> redirectStartPage. We don't actually expect a return in this scenario, but due to timing it does today.

Reply:
Another approach might to have `BrowserUtils.navigateWindow` return a promise and perform this delay

Reply:
Which would help prevent this bug from coming up in other scenarios.

Reply:
+1 to move the delay to navigateWindow if it's required. I also think we should consider making this configurable.